airflow/providers/google/cloud/operators/presto_to_gcs.py

```diff
@@ -0,0 +1,212 @@
+        Fetch the next row of a query result set, returning a single sequence, or
+        ``None`` when no more data is available.
+        """
+        if self.rows:
+            return self.rows.pop(0)
+        return self.cursor.fetchone()
+
+    def fetchmany(self, size=None):
+        """
+        Fetch the next set of rows of a query result, returning a sequence of sequences
+        (e.g. a list of tuples). An empty sequence is returned when no more rows are available.
+        """
+        if size is None:
+            size = self.cursor.arraysize
+
+        result = []
+        for _ in range(size):
+            row = self.fetchone()
+            if row is None:
+                break
+            result.append(row)
+
+        return result
+
+    def __next__(self):
+        """
+        Return the next row from the currently executing SQL statement using the same semantics as
+        ``.fetchone()``.  A ``StopIteration`` exception is raised when the result set is exhausted.
+        :return:
+        """
+        result = self.fetchone()
+        if result is None:
+            raise StopIteration()
+        return result
+
+    def __iter__(self):
+        """
+        Return self to make cursors compatible to the iteration protocol
+        """
+        return self
+
+
+class PrestoToGCSOperator(BaseSQLToGCSOperator):
+    """Copy data from PrestoDB to Google Cloud Storage in JSON or CSV format.
+
+    :param presto_conn_id: Reference to a specific Presto hook.
+    :type presto_conn_id: str
+    """
+
+    ui_color = "#a0e08c"
+
+    type_map = {
+        "BOOLEAN": "BOOL",
+        "TINYINT": "INT64",
+        "SMALLINT": "INT64",
+        "INTEGER": "INT64",
+        "BIGINT": "INT64",
+        "REAL": "FLOAT64",
+        "DOUBLE": "FLOAT64",
+        "DECIMAL": "NUMERIC",
+        "VARCHAR": "STRING",
+        "CHAR": "STRING",
+        "VARBINARY": "BYTES",
+        "JSON": "STRING",
+        "DATE": "DATE",
+        "TIME": "TIME",
+        # BigQuery don't time with timezone native.
+        "TIME WITH TIME ZONE": "STRING",
+        "TIMESTAMP": "TIMESTAMP",
+        # BigQuery supports a narrow range of time zones during import.
+        # You should use TIMESTAMP function, if you want have TIMESTAMP type
+        "TIMESTAMP WITH TIME ZONE": "STRING",
+        "IPADDRESS": "STRING",
+        "UUID": "STRING",
+    }
+
+    @apply_defaults
+    def __init__(self, presto_conn_id="presto_default", *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.presto_conn_id = presto_conn_id
+
+    def query(self):
+        """
+        Queries presto and returns a cursor to the results.
+        """
+        presto = PrestoHook(presto_conn_id=self.presto_conn_id)
+        conn = presto.get_conn()
+        cursor = conn.cursor()
+        self.log.info("Executing: %s", self.sql)
+        cursor.execute(self.sql)
+        return _AirflowPrestoCursorAdapter(cursor)
+
+    def field_to_bigquery(self, field):
```

Comment:
Would you mind adding type hints?

Reply:
Types are available in the base class. the mypy and other tools can identify types because they are abstract methods.

Reply:
Oh no.BaseSQLToGCSOperator has no type hints. I will add them in a separate PR.

Reply:
https://github.com/PolideaInternal/airflow/issues/738